### ISSUE TITLE
Problems with PDF annotations

### ISSUE THREAD

**Issue by HViethen85** [JabRef/jabref#2922]
Hi there, first of all I would have loved to use the issue tracker and automatic Report BUT there seams to be an error in the link directing to the automated issue report. The automated not working link ist:

https://github.com/JabRef/jabref/issues/new?title=Automatic+Bug+Report+-+20170617222228&body=JabRef+4.0.0-dev--snapshot--2017-06-17--master--c0fd08ec3%0D%0AWindows+10+10.0+amd64+%0D%0AJava+1.8.0_121%0A%0ASteps+to+reproduce%3A%0A%0A1.+...%0A2.+...%0A3.+...%0A%

That was the first issue. Now with the latest dev snapshot I have the following other issues

1. When more than one document is attatched I am not able to Switch between documents to retriev annotations. (Dropdown Menue does not work)
2. When more than one document ist attatched I cannot directly delete access files no longer needed via right mouse click delete local entry. In fact the only way I can do that is by deleting the file in the BibTex Key, then saving, then swopping between two entries, then going back to the original one and the access file is gone.

The next issue and probs one of the most annoying is, that often annotationspage does not fully build up. Also when using Jabref not in full window the Annotation window for highligts does not allow Scrolling passt a specific window size. It would also be a Suggestion to allow the user more flexibel to resize the annotations area for comments and Highlights. I use no comments but Highlights and it is a real pain at the current state because I often cannot read the full texts or the scroll bar gets completely locked despite the length of the text in the box.

Next issue. When there is trouble with the annotation area at the same time I am also not able to use the Status bar. However am able to use the varios Buttons and the program is otherwise functional.

Next issue, and that is one that makes the document viewer pretty useless. If there are highlights in the PDF document in the vewer instead of highlighting the text it is completely covered. The text can no longer be read.
JabRef+4.0.0-dev--snapshot--2017-06-17--master--   Windows10+10.0+amd64
Java+1.8.0_121

<details>  
Opening: C:\Users\Vieth\OneDrive\Abschicken\P-VEMA\VEMA.bib
Uncaught exception Occurred in Thread[JavaFX Application Thread,5,main]
java.lang.IllegalArgumentException: Cannot parse key binding SHIFT
	at javafx.scene.input.KeyCombination.valueOf(KeyCombination.java:429)
	at org.jabref.gui.keyboard.KeyBindingRepository.checkKeyCombinationEquality(KeyBindingRepository.java:81)
	at org.jabref.gui.keyboard.KeyBindingRepository.checkKeyCombinationEquality(KeyBindingRepository.java:170)
	at org.jabref.gui.keyboard.KeyBindingRepository.mapToKeyBinding(KeyBindingRepository.java:135)
	at org.jabref.gui.entryeditor.EntryEditor.lambda$setupKeyBindings$2(EntryEditor.java:183)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventFilterRecord.handleCapturingEvent(CompositeEventHandler.java:282)
	at com.sun.javafx.event.CompositeEventHandler.dispatchCapturingEvent(CompositeEventHandler.java:98)
	at com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(EventHandlerManager.java:223)
	at com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(EventHandlerManager.java:180)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchCapturingEvent(CompositeEventDispatcher.java:43)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:52)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$KeyHandler.process(Scene.java:3964)
	at javafx.scene.Scene$KeyHandler.access$1800(Scene.java:3910)
	at javafx.scene.Scene.impl_processKeyEvent(Scene.java:2040)
	at javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2501)
	at com.sun.javafx.tk.quantum.EmbeddedScene.lambda$null$300(EmbeddedScene.java:310)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.EmbeddedScene.lambda$keyEvent$301(EmbeddedScene.java:296)
	at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
	at java.lang.Thread.run(Unknown Source)

Uncaught exception Occurred in Thread[JavaFX Application Thread,5,main]
java.lang.IllegalArgumentException: Cannot parse key binding SHIFT
	at javafx.scene.input.KeyCombination.valueOf(KeyCombination.java:429)
	at org.jabref.gui.keyboard.KeyBindingRepository.checkKeyCombinationEquality(KeyBindingRepository.java:81)
	at org.jabref.gui.keyboard.KeyBindingRepository.checkKeyCombinationEquality(KeyBindingRepository.java:170)
	at org.jabref.gui.keyboard.KeyBindingRepository.mapToKeyBinding(KeyBindingRepository.java:135)
	at org.jabref.gui.entryeditor.EntryEditor.lambda$setupKeyBindings$2(EntryEditor.java:183)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventFilterRecord.handleCapturingEvent(CompositeEventHandler.java:282)
	at com.sun.javafx.event.CompositeEventHandler.dispatchCapturingEvent(CompositeEventHandler.java:98)
	at com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(EventHandlerManager.java:223)
	at com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(EventHandlerManager.java:180)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchCapturingEvent(CompositeEventDispatcher.java:43)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:52)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$KeyHandler.process(Scene.java:3964)
	at javafx.scene.Scene$KeyHandler.access$1800(Scene.java:3910)
	at javafx.scene.Scene.impl_processKeyEvent(Scene.java:2040)
	at javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2501)
	at com.sun.javafx.tk.quantum.EmbeddedScene.lambda$null$300(EmbeddedScene.java:310)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.EmbeddedScene.lambda$keyEvent$301(EmbeddedScene.java:296)
	at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
	at java.lang.Thread.run(Unknown Source)

</details>

**Comment by Siedlerchr** [comment]
Thanks for your report. I updated your post and removed the duplicate exception entries to make it more readable.

I can only comment on your second issue
The contextmenus for the file editors are coming back with the PR #2921 

For the first issue maybe @lynyus can help here.

**Comment by lynyus** [comment]
Hi @HViethen85,
It seems like you're the first user that is using the possibility to link several files to a BibEntry. (;
We are currently migrating large parts of the UI to a newer UI technology (JavaFX) so it might take a bit longer until these very valid issues that you have reported are fixed.
Would you mind attaching some screenshots to help us identifying all problems?

**Comment by HViethen85** [comment]
Yes of course, I will do so as soon as I am able to get to it. As I have discovered some more issues possibly relating.  It might take me a couple of days to do so however.

 

Von: Linus Dietz [mailto:notifications@github.com] 
Gesendet: Dienstag, 20. Juni 2017 16:56
An: JabRef/jabref
Cc: HViethen85; Mention
Betreff: Re: [JabRef/jabref] Problems with PDF annotations (#2922)

 

Hi @HViethen85 <https://github.com/hviethen85> ,
It seems like you're the first user that is using the possibility to link several files to a BibEntry. (;
We are currently migrating large parts of the UI to a newer UI technology (JavaFX) so it might take a bit longer until these very valid issues that you have reported are fixed.
Would you mind attaching some screenshots to help us identifying all problems?

**Comment by HViethen85** [comment]
So I finally managed to start taking screenshots of some of the described problem. 

Number one shows the strange placement and size of the “Highlight” Annotation field upon opening the annotations section for that document in full screen

 



 

The only way to have it change to a useable surface is by first downsizing the size of the jabref Window



 

and then resizing to full screen

 



 

Now this is also the entry with three chapters out of one book, each file individually loaded in ordert o have an easier overview over the annotations. In this mask it was possible to delete an additional entry by right mouse click delete file from database. However upon hitting the right mouse button nothing happens. The only way of deleting additional attached files you will be able to find below

 



 

Now swapping back to the annotations functions based on the design I would expect a drop down menu to open in order to allow me to swap files or for the annotations into the left field and as you can see, by exceeding one file the highlights area also has another hiccup.

 



 

Deleting additional files by first taking them out of the Bibtex Key 

 



If you now go back to the general setup after deletion you will still find the two files including the one deleted and also saving will not change that



 

Now you have to simply go on a different jabref entry then go back to the one you deleted the file from and it is gone

 



 

An additionally you see the next hiccup by jabref now deciding that the file Window size needs a change in height, even when nothing has changed to the setup.

 

Now I chose a different entry and this time I cannot even see the area of the highlights in full screen

 



 

After resizing the window as mentioned above it is a workable surface however still lacking the option to copy to clipboard. And since I am not able to copy anything out of Jabref anymore that is also not very functional. This also includes Copy and Paste for example if you have mixed up Authors and Title and want to resolve by copy and paste or even worse cut and paste as that will paste it directly into the search area instead.

 



 

As I am in the midst of some exams I have to report as I go along. However I would also be willing to offer the option of one developer to take a direct look at the problems in real live via a tool like Teamviewer as some of the issues are hard to pick up in a screenshot. However, I hope this is able to highlight the issues I am experiencing better.

 

Heike 

 

 

Von: Linus Dietz [mailto:notifications@github.com] 
Gesendet: Dienstag, 20. Juni 2017 16:56
An: JabRef/jabref
Cc: HViethen85; Mention
Betreff: Re: [JabRef/jabref] Problems with PDF annotations (#2922)

 

Hi  <https://github.com/hviethen85> @HViethen85,
It seems like you're the first user that is using the possibility to link several files to a BibEntry. (;
We are currently migrating large parts of the UI to a newer UI technology (JavaFX) so it might take a bit longer until these very valid issues that you have reported are fixed.
Would you mind attaching some screenshots to help us identifying all problems?

**Comment by lynyus** [comment]
@HViethen85 For some reason the screenshots are not visible for me.

**Comment by HViethen85** [comment]
@lynyus I was not aware, that they could not be uploaded via Outlookresponse. For some reason adding them on to the message does  not help, as I myself am not able to open them. I will try placing them in this message and hope that this time it might work. For some reason they remain not accessible.
Here are the alternate links in order as mentioned above:

Screen 1
https://1drv.ms/i/s!AjUo3r3jr7AIg_AVVtqEh2e8-gY11A
Screen 2
https://1drv.ms/i/s!AjUo3r3jr7AIg_AWj8-GMHsiVf0TGA
Screen 3
https://1drv.ms/i/s!AjUo3r3jr7AIg_AXGDbx9AW_yro9IA
Screen 4
https://1drv.ms/i/s!AjUo3r3jr7AIg_AYr61JdnpyDn_k7g
Screen 5
https://1drv.ms/i/s!AjUo3r3jr7AIg_AZBRycvt6LOKDB0w
Screen 6
https://1drv.ms/i/s!AjUo3r3jr7AIg_Aag6XHIgS0p1QbyA
Screen 7
https://1drv.ms/i/s!AjUo3r3jr7AIg_AbPQV7vOO9S6sHUQ
Screen 8
https://1drv.ms/i/s!AjUo3r3jr7AIg_AciHDt98ZFe5kBTg
Screen 9
https://1drv.ms/i/s!AjUo3r3jr7AIg_AdIqfaaKvUp-X17g
Screen 10
https://1drv.ms/i/s!AjUo3r3jr7AIg_AeUnFbd-TrHXn5Rw
Screen 11
https://1drv.ms/i/s!AjUo3r3jr7AIg_AfQfSl4Ykw0f0gQw

**Comment by HViethen85** [comment]
And as I go along, there is yet the next Problem. 

1.       Entrytype “Techreport” 

2.       Optional fields

3.       Type

4.       Defining a new Techreport Type by typing it in 

5.       Nothing happens, when closing the entry or  swopping between register the entried data is lost and not ingetrated into the entry

In the previous Version I was able to define my own techreport types simply by typing them in, as some of the types I need in my area of study are not mentioned or need to be mentioned in a different way. So far the only way (which I was able to figure out) to assure the correct Type is defined, is by first choosing a wrong type via the dropdown, the going to the BibTex area, there changing the wrong type to the correct type, then swopping between two entries, going back to the one just edited and then you have the correct entry. That however is quite time consuming.

 

I also noticed, that even though I set up the language to being German, whilst the top registers are displayed in German, not are the field information titles.

 

So long,

Heike
